### DISCUSSION TITLE
`!<command>` does not work lazer client chat

### DISCUSSION THREAD

**Discussion by msmcculley** [ppy/osu#21967] [Q&A] [answered]
Pretty much the title: 

Tried sending multiple !report commands and nothing went through. Decided to test using !faq commands and it seems to have regressed? Before 2022.2812-0 (from my knowledge) !<commands> were usable.

![](https://media.discordapp.net/attachments/312507967570444288/1058615331771392101/image.png)

<img width="1082" alt="image" src="https://user-images.githubusercontent.com/90729492/210126074-433c44dc-2e68-4431-81b5-c302baaadc50.png">

Edited for log purposes: 

[database.log](https://github.com/ppy/osu/files/10326851/database.log)
[input.log](https://github.com/ppy/osu/files/10326852/input.log)
[network.log](https://github.com/ppy/osu/files/10326853/network.log)
[performance.log](https://github.com/ppy/osu/files/10326854/performance.log)
[runtime.log](https://github.com/ppy/osu/files/10326855/runtime.log)

**Comment by bdach** [comment]
There is nothing in the latest releases that should have regressed this. Those commands were never handled client side. I'm not sure why commands starting with `!` are seemingly handled client side either. Looking at source only the ones starting with `/` should be.

**Reply by msmcculley** [reply]
I looked back to when I tested it previously and it was back in October 2021, where !<command> was actually usable. 

Regressed might've not been the right terminology.

**Reply by msmcculley** [reply]
For reference this is 2021.1016-0: ![image](https://user-images.githubusercontent.com/90729492/210132868-c03b445b-a900-4fb9-a792-c082f4fc152d.jpeg)

**Reply by bdach** [reply]
The closest think that I can think of that may have regressed this is the transition to receiving messages via websocket (https://github.com/ppy/osu/pull/20991). And even that didn't change how messages are *sent*, so I think there is a chance that the messages are just displayed back wrong?

Can you see if you see the same message content when you check via website, or via stable client?

**Reply by msmcculley** [reply]
Stable client acts as it normally does. Getting a message response back when using !report and when doing !faq or any `!` in a chat will redirect to banchobot with the displayed information for said command.

On website neither `/` commands work or `!`

**Comment by msmcculley** [comment]
I'm guessing this is just going to be another "not important enough" type issue then. Despite now there being no way for players to report people solely in lazer, let alone use Bancho's other commands for like multiplayer, refereeing, etc?

**Comment by peppy** [comment, marked as answer]
This is a bit weird. Messages are supposed to be sent to bancho in a way that it can reply to these commands for backwards compatibility. I'll take a look into it.

**Reply by frenzibyte** [reply]
#22083
